log/log.go

```diff
@@ -0,0 +1,71 @@
+package log
+
+import (
+	"log"
+	"os"
+	"strings"
+	"testing"
+)
+
+// Logger defines the Logger interface
+type Logger interface {
+	Printf(format string, v ...any)
+}
+
+// defaultLogger is the default Logger instance
+var defaultLogger Logger = &noopLogger{}
+
+func init() {
+	// Enable default logger in the testing with a verbose flag.
+	if testing.Testing() {
+		// Parse manually because testing.Verbose() panics unless flag.Parse() has done.
+		for _, arg := range os.Args {
+			if strings.EqualFold(arg, "-test.v=true") || strings.EqualFold(arg, "-v") {
+				defaultLogger = log.New(os.Stderr, "", log.LstdFlags)
+			}
+		}
+	}
+}
+
+// Default returns the default Logger instance
+func Default() Logger {
+	return defaultLogger
+}
+
+// SetDefault sets the default Logger instance
+func SetDefault(logger Logger) {
+	defaultLogger = logger
+}
+
+func Printf(format string, v ...any) {
+	defaultLogger.Printf(format, v...)
+}
+
+// Validate our types implement the required interfaces.
+var (
+	_ Logger = (*log.Logger)(nil)
+)
```

Comment:
suggestion: can we move this to the top of the file add add one for each logger implementation